Kickstarter-iOS/Features/Comments/Controllers/CommentRepliesViewController.swift

```diff
@@ -159,18 +169,35 @@ final class CommentRepliesViewController: UITableViewController {
           self?.tableView.scrollToRow(at: indexPath, at: .top, animated: false)
         }
       }
+
+    self.viewModel.outputs.userBlocked
+      .observeForUI()
+      .observeValues { [weak self] success in
+        self?.commentComposer.isHidden = true
+
+        if success {
+          self?.messageBannerViewController?
+            .showBanner(with: .success, message: "This user has been successfully blocked")
+          self?.view.isUserInteractionEnabled = false
```

Comment:
I suspect that this behavior should be part of `MessageBannerViewController`. Having a banner that blocks interaction seems like a fairly common pattern. You could also add (if it doesn't already exist) a touch gesture that dismisses the banner when any additional area is tapped.

I peeked in `MessageBannerViewController` and it is an older-style class, which is probably way out of scope for any substantial edits, but it might be worthwhile folding in just this bit while you're in there.

Reply:
This is interesting. It would require some reworking of `MessageBannerViewController` to hold on to a reference to the parent view, but that might end up being less involved and we might be able to remove the delegate method altogether. I'll look into it.

a touch gesture that dismisses the banner already does exists 👍

Reply:
actually - since this banner is being used in other places in the app I'd rather address any refactoring of this class in a separate PR. This functionality is specific to the blocking flow currently as well and I'd rather not risk introducing regressions elsewhere.

Reply:
I think that's fair.